xlsxwriter/worksheet.py

```diff
@@ -376,58 +413,45 @@ def write(self, row, col, *args):
 
         # Write None as a blank cell.
         if token is None:
-            return self.write_blank(row, col, *args)
+            return self._write_blank(row, col, *args)
+
+        # avoid isinstance as much as possible (because it's slow)
```

Comment:
I don't the duplication this change causes. Is the performance increase worth it?

Reply:
True. It is duplication, but only because I wanted to leave support for developers who might be extending the base classes with their own craziness (people tend to do this with strings sometimes).  Checking for type is _SO_ fast that a single isinstance call would overwhelm it.  please see: http://canonical.org/~kragen/isinstance/

I did a similar fix for the json decoder in Django (unfortunately this was not open source) when I was working for a company fielding 3000 requests per _second_ on a slow day.

This change made a huge difference for my current codebase, even though my current codebase hits the "isinstance" block frequently since we are rendering numpy arrays.  I _think_ I could make this faster also.

btw, why not support numpy more natively if it's installed (with try: ... except ImportError), seems like a no brainer for a project that's writing spreadsheets.